README.md

```diff
@@ -3,65 +3,175 @@ Manalize
+
+### Setting up your environment
+
+Given you have a web project that you clone for the first time and you need to run locally, just use the `setup` command:
+
+```sh
 $ manalize setup ~/my-awesome-project
 ```
 
-##### Metadata only
+This command interactively configures the virtual machine for your project.  
+Once this step done, your environment is fully configured. Some files have been added to your project:
 
-Sometimes, it can be handy to reconfigure the environment(s), without affecting the project files (in order to use the diff command with a new environment config for instance).
+- A `Vagrantfile`
+- A `Makefile` including some useful tasks that you'll need to use throughout your project
+- An `ansible/` directory containing all the configuration related to the VM provisioning
+- Eventually some files specific to the chosen environment
 
-The following command will only update the `ansible/.manalize` metadata file:
+At this point, the VM can be provisioned using the following command:
 
+```sh
+$ make setup
 ```
-$ manalize setup --no-update ~/my-awesome-project
+
+### Working with your environment
+
+Once the [`setup`](#setting-up-your-environment) process is finished (may take a few minuts), your environment is operational and your VM is running.  
+To manage it and work with, just use the `vagrant` command-line tool as usual:
+
+```sh
+$ vagrant up/halt/reload/ssh
+```
+
+### Keeping your environment up-to-date
+
+Given your project's environment is there and your VM works well, its configuration is sticked to what we provided at the moment you created it.  
+Since the [manala ansible roles](http://manala.io/) evolve (and the corresponding templates as well), you may want to be aware of the additions and important changes made to, then update your environment accordingly.
+
+To do so, there's two commands to be aware of: `self-update` and `diff`.
+
+#### self-update
+
+```sh
+$ manalize self-update
 ```
 
-#### Diff
+Running this command updates your `manalize` binary to the latest release, coming with the latest configuration templates.
+After that, you can safely use the `diff` command as shown below.
+
+#### diff
 
-Get the diff:
+The `diff` command allows you to get a patch representing the diff between your current project configuration and the ones that would have been provided by your current version of the `manalize` binary.
+
+Getting the diff:
 ```
 $ manalize diff ~/my-awesome-project
 ```
 
-Apply the diff:
+Getting the diff for applying the patch immediately:
 ```
 $ cd ~/my-awesome-project
 $ manalize diff | git apply
 ```
 
+Getting the diff for applying the patch later:
+```
+$ cd ~/my-awesome-project
+$ manalize diff > manalize.patch
+$ git apply manalize.patch
+```
+
+_Note:_
+  
+Be careful when applying the patch, any custom change made to your environment configuration will be erased.  
+To minimize risks, we recommend you to look at the patch before trying to apply it. 
```

Comment:
See http://nuclearsquid.com/writings/git-add/

maybe we can explain how to proceed (and mention PHPStorm VCS > Apply Patch which allow to apply it interactively)